src/lib/krb5/os/trace.c

```diff
@@ -173,7 +173,7 @@ trace_format(krb5_context context, const char *fmt, va_list ap)
             p = va_arg(ap, const char *);
             if (p == NULL && len != 0)
                 k5_buf_add(&buf, "(null)");
-            else
+            else if (p != NULL)
```

Comment:
No, if p is NULL and len is 0, that's not a failure case; we don't want to display "(null)" in this case, and functionally it doesn't matter if we do nothing or call buf_add_printable_len(&buf, NULL, 0) as that is a no-op.

It's sort of annoying that Coverity trips over this code, but adding the conditional isn't too bad as it just looks like an optimization for the p == NULL && len == 0 case.